chain/src/chain.rs

```diff
@@ -503,7 +502,9 @@ impl ChainService {
             if log_enabled!(ckb_logger::Level::Debug) {
                 self.print_chain(10);
             }
-            metrics!(gauge, "ckb.chain_tip", block.header().number() as i64);
+            if ckb_metrics::metrics_enabled() {
+                ckb_metrics::CKB_CHAIN_TIP.set(block.header().number() as i64);
```

Comment:
Can we enforce that we must check before recording?

Reply:
Solution 1: using macro, and disable the direct access to meters like `ckb_metrics::CKB_CHAINTIP`

Solution 2: use a proxy object to wrap meters like `ckb_metrics::CKB_CHAINTIP`. And the `set`, `inc` functions in the proxy objects will check whether metrics are enabled. However, if we don't use metrics, args are still evaluated.

Reply:
I'd like to choose Solution 2.

Reply:
Let's just keep the `if enabled? { do_metric() }` explicit in the code.

It's simple and we don't really care much the performance lose when we forget to add the `if guard`.

Reply:
I have implement solution 2, but I forget push my commit yesterday. :sob: